src/EnergyPlus/SimAirServingZones.cc

```diff
@@ -1441,107 +1441,50 @@ void InitAirLoops(EnergyPlusData &state, bool const FirstHVACIteration) // TRUE
-    Real64 MassFlowSaved;         // mass flow rate for a node saved from previous call
-    Real64 MassFlowSet;           // desired mass flow rate for a node
-    int SupAirPath;               // supply air path do loop index
-    int SupAirPathNum;            // specific supply air path index
-    int SplitterNum;              // Zone equip splitter index
-    int PlenumNum;                // supply plenum index
-    int CtrlZoneNum;              // Controlled zone index
-    int ZoneInNum;                // zone inlet index
-    int ZoneInSysIndex;           // index into CoolCtrlZoneNums or HeatCtrlZoneNums
-    int NumComponentsInSys;       // total number of components in the primary air system
-    int NumComponentsOnBranch;    // total number of components in the primary air system
-    bool FoundSupPathZoneConnect; // true if there is a valid connection between the supply air path and a zone terminal unit inlet
-    CompType CompTypeNum;         // component_type number for components on branches
-    int SupFanIndex;
-    int RetFanIndex;
-    bool FoundOASys;
-
-    // Dimension the local subcomponent arrays
+    int const numPrimaryAirSys = state.dataHVACGlobal->NumPrimaryAirSys;
 
-    auto &NumPrimaryAirSys = state.dataHVACGlobal->NumPrimaryAirSys;
-
-    bool ErrorsFound;
-    Array1D_int tmpNodeARR;
-    int nodeLoop;
-    int ZoneNum;
-
-    ErrorsFound = false;
+    bool ErrorsFound = false;
     state.dataHVACGlobal->AirLoopInit = true;
 
-    auto &PrimaryAirSystems(state.dataAirSystemsData->PrimaryAirSystems);
-    auto &AirToZoneNodeInfo(state.dataAirLoop->AirToZoneNodeInfo);
-    auto &AirLoopControlInfo(state.dataAirLoop->AirLoopControlInfo);
-
     // Do the one time initializations
     if (state.dataSimAirServingZones->InitAirLoopsOneTimeFlag) {
 
         // Figure out what zones are served by each primary air system (air loop) and
         // store the results in AirToZoneNodeInfo()%CoolCtrlZoneNums and AirToZoneNodeInfo()%HeatCtrlZoneNums
 
-        // Allocate scratch arrays for storing controlled zone numbers for each air loop.
-        state.dataSimAirServingZones->CtrlZoneNumsCool.allocate(state.dataGlobal->NumOfZones);
-        state.dataSimAirServingZones->CtrlZoneNumsHeat.allocate(state.dataGlobal->NumOfZones);
-        state.dataSimAirServingZones->ZoneInletNodesCool.allocate(state.dataGlobal->NumOfZones);
-        state.dataSimAirServingZones->ZoneInletNodesHeat.allocate(state.dataGlobal->NumOfZones);
-        state.dataSimAirServingZones->TermInletNodesCool.allocate(state.dataGlobal->NumOfZones);
-        state.dataSimAirServingZones->TermInletNodesHeat.allocate(state.dataGlobal->NumOfZones);
-        state.dataSimAirServingZones->TermUnitSizingNumsCool.allocate(state.dataGlobal->NumOfZones);
-        state.dataSimAirServingZones->TermUnitSizingNumsHeat.allocate(state.dataGlobal->NumOfZones);
+        // Temporary air loop zone data
+        struct AirloopZone
```

Comment:
Per @amirroth suggestion, combined 8 arrays into two vectors of this struct.

Reply:
Very nice refactor.